### ISSUE TITLE
Failing to clusterctl  init with kubevirt provider

### ISSUE THREAD

**Issue by galbeniluz** [kubernetes-sigs/cluster-api-provider-kubevirt#147]
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]
I'm getting this error from clusterctl when trying to run clusterctl init with Kubevirt as a provider 
`clusterctl init --infrastructure kubevirt
Fetching providers
Error: failed to get provider components for the "kubevirt" provider: failed to get repository client for the InfrastructureProvider with name kubevirt: error creating the GitHub repository client: failed to get GitHub latest version: failed to find releases tagged with a valid semantic version number`

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version: 1.1.15
- Cluster-api-provider-kubevirt version: latest
- Kubernetes version: (use `kubectl version`): 1.22.10
- KubeVirt version: 0.54.0
- OS (e.g. from `/etc/os-release`):centos

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/labels?q=area for the list of labels]

**Comment by isaacdorfman** [comment]
This is probably due to the clusterctl not accepting v0.1.0-rc.0 as a valid semver version.

@davidvossel @agradouski 
Can you create and push the tag v0.1.1 for the same commit, to trigger a release with a semver version accepted by clusterctl?

**Comment by davidvossel** [comment]
@agradouski @cchengleo could one of you create and push the tag? I don't have permissions


```
git tag -s "v0.1.0" -m "v0.1.0 release"
git push origin v0.1.0
```

**Comment by galbeniluz** [comment]
@davidvossel @cchengleo Thanks for the fix but I'm still getting the same error.
I think that the issue is that the metadata.yaml contains a 1.1 release which doesn't exist
https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/blob/3fbdad97b90f6dd4cc691a213f2d86f992d39937/metadata.yaml#L9

**Comment by galbeniluz** [comment]
`❯ clusterctl init --infrastructure kubevirt:v0.1.0 -v5          
Using configuration File="/Users/gal.beniluz/.cluster-api/clusterctl.yaml"
Fetching providers
Fetching File="core-components.yaml" Provider="cluster-api" Type="CoreProvider" Version="v1.1.5"
Fetching File="bootstrap-components.yaml" Provider="kubeadm" Type="BootstrapProvider" Version="v1.1.5"
Fetching File="control-plane-components.yaml" Provider="kubeadm" Type="ControlPlaneProvider" Version="v1.1.5"
Fetching File="infrastructure-components.yaml" Provider="kubevirt" Type="InfrastructureProvider" Version="v0.1.0"
Fetching File="metadata.yaml" Provider="cluster-api" Type="CoreProvider" Version="v1.1.5"
Fetching File="metadata.yaml" Provider="kubeadm" Type="BootstrapProvider" Version="v1.1.5"
Fetching File="metadata.yaml" Provider="kubeadm" Type="ControlPlaneProvider" Version="v1.1.5"
Fetching File="metadata.yaml" Provider="kubevirt" Type="InfrastructureProvider" Version="v0.1.0"
Error: invalid provider metadata: version v0.1.0 for the provider capk-system/infrastructure-kubevirt does not match any release series
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.(*providerInstaller).getProviderContract
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/installer.go:248
sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster.(*providerInstaller).Validate
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster/installer.go:204
sigs.k8s.io/cluster-api/cmd/clusterctl/client.(*clusterctlClient).Init
        sigs.k8s.io/cluster-api/cmd/clusterctl/client/init.go:110
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.runInit
        sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/init.go:149
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.glob..func10
        sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/init.go:88
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.2.1/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.2.1/command.go:974
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.2.1/command.go:902
sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.Execute
        sigs.k8s.io/cluster-api/cmd/clusterctl/cmd/root.go:94
main.main
        sigs.k8s.io/cluster-api/cmd/clusterctl/main.go:26
runtime.main
        runtime/proc.go:255
runtime.goexit
        runtime/asm_arm64.s:1133
`

**Comment by hh** [comment]
Noticing this as well, is a way to override, create a local work around until the the referenced infrastructure provider yaml / release is updated?

**Comment by davidvossel** [comment]
@nunnatsa is this something you'd be able to investigate?

**Comment by isaacdorfman** [comment]
@davidvossel I think that changing the major version in metadata.yaml from 1 to 0 as a quick fix and running the CI will fix it.

**Comment by davidvossel** [comment]
@hh @galbeniluz This should be resolved now with the new v0.1.1 capk release

**Comment by galbeniluz** [comment]
@davidvossel Its working now
Thank you
